lib/stages/N2TimeDownsample.cpp

```diff
@@ -0,0 +1,340 @@
+            output_frame.eop = eop_target;
+
+            if (do_fringestop) {
+                // Get the per-dish fringestopping phases.
+                tel.fringestop_phases_1d(freq_Hz, frame.eop, eop_target, fringe_phase);
+
+                // This indexing requires the el_id = (n_dish)*pol_id + dish_id
+                size_t idx = 0;
+                for (size_t i = 0; i < num_elements; i++) {
+                    for (size_t j = i; j < num_elements; j++) {
+
+                        // To apply phases:
+                        //  Fringestop(V_ij) = V_ij * exp{i*(phi_i - phi_j)}
+                        //                   = V_ij * Phase_i * conj(Phase_j)
+
+                        size_t d_i = i % num_dishes;
+                        size_t d_j = j % num_dishes;
+                        output_frame.vis[idx] *= fringe_phase[d_i] * std::conj(fringe_phase[d_j]);
+
+                        idx++;
+                    }
+                }
+            }
+
+            for (size_t i = 0; i < nprod; i++) {
+                output_frame.weight[i] = 1. / output_frame.weight[i];
+            }
+
+            for (uint32_t i = 0; i < num_eigenvectors; i++) {
+                for (uint32_t j = 0; j < num_elements; j++) {
+                    // Eigenvectors get phases too.
+                    int k = i * num_elements + j;
+                    size_t d_j = j % num_dishes;
+                    output_frame.evec[k] *= fringe_phase[d_j];
+                }
+            }
+
+            // Go to next frame
+            nframes += 1;
+            in_buf->mark_frame_empty(unique_name, frame_id++);
+            continue;
+        }
+
+        // If we're here, an accumulation has started.
+
+        auto output_frame = N2FrameView(out_buf, output_frame_id);
+
+        // Check we are still in accumulation window
+        if (frame.eop.ERA_deg >= era_deg_lo && frame.eop.ERA_deg < era_deg_hi) {
+
+            // Recalculate fringestop phases
+            if (do_fringestop)
+                tel.fringestop_phases_1d(freq_Hz, frame.eop, eop_target, fringe_phase);
+
+            // Accumulate contents of buffer
+            size_t idx = 0;
+            for (size_t i = 0; i < num_elements; i++) {
+                for (size_t j = i; j < num_elements; j++) {
+
+                    size_t d_i = i % num_dishes;
+                    size_t d_j = j % num_dishes;
+
+                    // Computing the total phase in double precision
+                    // in case one of the dish phases is small.
+                    std::complex<double> w_doub = fringe_phase[d_i] * std::conj(fringe_phase[d_j]);
+
+                    // Now truncate the phase to a float to match vis[]
+                    // Have to be explicit about this, compiler complains
+                    // otherwise.
+                    N2::cfloat w_fs{(float)w_doub.real(), (float)w_doub.imag()};
+
+                    // Accumulate
+                    output_frame.vis[idx] += w_fs * frame.vis[idx];
```

Comment:
```suggestion
                    output_frame.vis[idx] += w_fs * frame.n_valid_fpga_ticks_in_frame * frame.vis[idx];
```
Should we weight `frame.vis` by the number of sub-integrations that were averaged together to generate it? I believe this is the `frame.n_valid_fpga_ticks_in_frame` quantity?  Otherwise, a "complete" frame and a frame that is heavily masked due to RFI or packet loss would be given equal weight, even though the latter could be significantly noisier.

Reply:
This would change how the `vis` dataset is normalized below, and how the `weight`, `eval`, `evec`, and `erms` quantities are accumulated and normalized.

Reply:
Correct!  This is addressed in commit b2a03ef .